### ISSUE TITLE
usage with jest / esm / typescript

### ISSUE THREAD

**Issue by ChrisKretschmer** [open-wc/open-wc#2266]
hey all,

I'm currently trying to use @open-wc/testing or @open-wc/semantic-dom-diff in an environment with ts-jest in a project, where type="module" is set in the package.json.

When I try to use any package from @open-wc, I have the issue, that jest dies with the usual error message:

```
Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation
```

Changing transformIgnorePatterns does not make a difference. The problem is, that the files from @open-wc are treated as a commonjs module by jest because type: "module" is not set in the package.json.

If I change the package.json in the @open-wc libraries to include type: "module", everything works perfectly.

Is there a reason, that this is not included there? This breaks the compatibility with nearly every testing framework, if you use esm in your project.

## Expected behavior

the @open-wc packages should be usable for testing in projects using esm and typescript with jest / jasmine / mocha / tap.
(I tried them all...)

## Actual Behavior

Tests can't run, because includes are treated as commonJS packages - They break because of the import statements

## Additional context

Which packages and versions are you using.

```
"dependencies": {

    "lit": "^2.0.0-rc.2"
  },
  "devDependencies": {
    "@11ty/eleventy": "^0.12.1",
    "@11ty/eleventy-plugin-syntaxhighlight": "^3.0.1",
    "@open-wc/semantic-dom-diff": "^0.19.4",
    "@open-wc/testing": "^2.5.33",
    "@types/eslint": "^7.28.0",
    "@typescript-eslint/eslint-plugin": "^4.29.0",
    "@typescript-eslint/parser": "^4.29.0",
    "@web/dev-server": "0.1.20",
    "@web/dev-server-legacy": "^0.1.4",
    "@webcomponents/webcomponentsjs": "^2.6.0",
    "copyfiles": "^2.4.1",
    "deepmerge": "^4.2.2",
    "eslint": "^7.32.0",
    "lit-analyzer": "^0.0.6",
    "prettier": "^2.3.2",
    "rimraf": "^3.0.2",
    "ts-loader": "^9.2.5",
    "typescript": "^4.3.5",
    "web-component-analyzer": "^1.2.0-next.0",
    "webpack": "^5.48.0",
    "webpack-cli": "^4.7.2",
    "@types/jest": "^26.0.24",
    "cross-env": "^7.0.3",
    "esm": "^3.2.25",
    "jest": "^27.0.6",
    "ts-jest": "^27.0.4",
    "ts-jest-resolver": "^1.1.0",
    "ts-node": "^10.1.0"
  }
```

**Comment by LarsDenBakker** [comment]
We can add them, thats not a problem. 

These packages are made to run in the browser, for example using web test runner or karma, so we cannot guarantee they will work in a node environment. This is also why type module is not there, since commonjs cant work in the browser

**Comment by ChrisKretschmer** [comment]
I had some issues with web test runner and karma (namely getting esm and typescript to work), so i was happy to find that jest was working in that environment...

Should I create a PR for this?

**Comment by ChrisKretschmer** [comment]
#2267 

I created a pull request for all of the esm packages

**Comment by pmcelhaney** [comment]
Just ran into the same problem / solution. Delighted to see that there's already a pull request!

**Comment by pmcelhaney** [comment]
@LarsDenBakker Any update on when the pull request can be merged?

**Comment by jdvivar** [comment]
Someone with admin rights please close this issue, please? The PR is merged but wasn't linked to the issue

**Comment by Westbrook** [comment]
Fixed in #2267
